bgpd/bgp_routemap.c

```diff
@@ -1441,7 +1441,8 @@ route_set_evpn_gateway_ip(void *rule, const struct prefix *prefix, void *object)
 
 	/* Set gateway-ip value. */
 	bre->type = OVERLAY_INDEX_GATEWAY_IP;
-	memcpy(&bre->gw_ip, &gw_ip->ip.addr, IPADDRSZ(gw_ip));
+	bre->gw_ip.ipa_type = gw_ip->ipa_type;
```

Comment:
hmm: aren't both of these struct ipaddr? why not just copy the whole struct (like with `=`)?

Reply:
I thought about it. Matter of taste I guess, I preferred to do it this way rather than the following:

```c
	memcpy(&bre->gw_ip, gw_ip, sizeof(enum ipaddr_type_t) + IPADDRSZ(gw_ip));
```

I can change it if you want. I find the two copies kinda more readable (the real foolproof way to avoid other issues like this would be to have an IP copy function).

Most places I could find in FRR setting the IP first set the IP type then memcpy the contents, so I think it's better to keep the same pattern (copy a "simple" variable, memcpy the variable length part).

Reply:
ah sorry, I misunderstood your comment. I didn't read your "like with `=`" as "use `=`". Yes, that is definitely possible, though I wonder why it's not more widely used in FRR.

Changed to a plain `=`.